src/main/java/adris/altoclef/StaticMixinHookups.java

```diff
@@ -1,57 +1,57 @@
 package adris.altoclef;
 
+
 import adris.altoclef.commandsystem.CommandException;
 import baritone.api.event.events.ChatEvent;
 import net.minecraft.block.BlockState;
 import net.minecraft.client.gui.screen.Screen;
 import net.minecraft.client.util.math.MatrixStack;
-import net.minecraft.entity.LivingEntity;
 import net.minecraft.entity.player.PlayerEntity;
-import net.minecraft.item.ItemStack;
 import net.minecraft.util.hit.BlockHitResult;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.ChunkPos;
 import net.minecraft.world.World;
 import net.minecraft.world.chunk.WorldChunk;
-import org.jetbrains.annotations.Nullable;
+
 
 /**
  * Mixins have no way (currently) to access our mod.
- *
+ * <p>
  * As a result I'll do this statically.
- *
- * However, I want to avoid grabbing AltoClef in a static context, so this class
- * serves at the "static dumpster" which is the only spot where
- * I allow the bad practice of singletons to flourish
+ * <p>
+ * However, I want to avoid grabbing AltoClef in a static context, so this class serves at the "static dumpster" which is the only spot
+ * where I allow the bad practice of singletons to flourish
  */
 public class StaticMixinHookups {
+    // TODO: 2021-05-22 would be much nicer  to have a static `getInstance()` method in `AltoClef`.
```

Comment:
I see you're trying to made my code more readable and well designed.

But what you're suggesting actually encourages the singleton anti-pattern.

![lol](https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fi.ytimg.com%2Fvi%2F6bI9Rg9l100%2Fmaxresdefault.jpg&f=1&nofb=1)

Reply:
Yes, because it essentially _is_ a singleton.

It should only ever be instantiated once: when minecraft is started.

Reply:
Hmm yes, it definitely is basically a singleton now, but I'd rather leave that open.

There's a reason why you see `someFunction(AltoClef mod, ...)` everywhere, as it's passed down through functions. I don't want to make any assumptions here: In the far future I may have multiple bots running from the same program somehow, and each one may have its own version of AltoClef running.

I'm not saying that by turning AltoClef into a singleton I'll necessarily _have_ to use it statically everywhere, but I'd rather not leave that bad practice available. It rubs me the wrong way I guess.

Also, I'm containing all of the "necessary static stuff" into one class, "StaticMixinHookups". May as well keep that style that I want to avoid contained in one file.

Reply:
I'm gonna be honest, I doubt that there'd be a nice way to do that. And if you DID do it, it would work something like this:
you have a "coordinator" program which communicates via sockets or smth to the other programs, and is run independently of minecraft.